index.js

```diff
@@ -243,19 +278,26 @@ class LatticeKeyring extends EventEmitter {
   signMessage(address, msg) {
     return new Promise((resolve, reject) => {
       this._unlockAndFindAccount(address)
```

Comment:
In both calls to `_unlockAndFindAccount`, the initial logic within subsequent `then` is the same. Should that initial logic be merged into the `Promise` that gets returned?

Reply:
Specifically:
```
         // Make sure the account is associated with the current wallet
         if (this.accountOpts[accountIdx].walletUID !== this._getCurrentWalletUID()) {
           return reject(new Error('Account associated with different wallet. ' +
                                   'Please switch to this wallet on your Lattice to continue.'));
         }
```

Reply:
5666dd5